### DISCUSSION TITLE
Blacklist One (or more) Monitors Unless Key Pressed

### DISCUSSION THREAD

**Discussion by cevans3505** [Eliasyoussef47/LockCursorInMonitor#11] [Ideas]
I have a 4 monitor setup where 3 are in a row horizontally and 1 up above which maintains a static dashboard which I monitor during work, but have no need to actively manipulate. Sometimes it'll catch my mouse when moving stuff around monitors 1-3, so I'm looking for a way to blacklist the mouse from monitor 4 unless a specific key (or any way to toggle it) is pressed.

![Screenshot 2021-03-22 190449](https://user-images.githubusercontent.com/22597403/112069180-c2aa2e00-8b41-11eb-9553-3b6a440048cb.png)

**Comment by Eliasyoussef47** [comment]
I'll look into it. This might even need a completely new program but I'll see what I can do. I'm busy with exams these two weeks so I don't think I'll be able to accomplish much this period.

Sounds like a cool challenge. Thank you for the idea.

**Reply by cevans3505** [reply]
Hey man, do your thing. I just stumbled across your repository from a separate link (I think StackOverflow) and saw that it was actively maintained and thought I'd throw out the idea since I've not been successful in finding any currently available solutions (aside from somewhat hacky AutoHotKey scripts which I'm not opposed to, but I would prefer a more elegant solution). If you're up for the challenge and come up with something that'd be awesome. Best of luck on your exams 😁

**Reply by Eliasyoussef47** [reply]
There is one question I'd like to ask you. Have you tried this program (LockCursorInMonitor)? if so, does it work?
If you haven't tried it yet it would be useful if you test it for me on your machine to see if it's compatible. This would be useful if I'll build a new program.

**Reply by cevans3505** [reply]
I hadn't tried your LockCursorInMonitor, but I've tried several a handful of similar tools. Most notably WindowedBorderlessGaming which I often used to lock my mouse to a particular monitor while running a game in "Borderless Windowed" mode. I just downloaded it though and it runs just fine. I'm able to keep my mouse bound to any particular monitor so long as I'm holding down Ctrl. I'm currently running Windows 10 Pro for Workstations (21H1) if that makes any difference. I ended up resorting to running one of my Raspberry Pi's connected to my 4th monitor to simply display the dashboard I need to keep up there for now that way my mouse doesn't get snagged by it. I'd obviously like to be able to keep everything on my PC so I can free up my Pi for other projects, haha, but it's my temporary solution for now.

**Reply by Eliasyoussef47** [reply]
Got it. Thanks for your help. I'll definitely try to make a solution for you.

**Reply by Eliasyoussef47** [reply]
@cevans3505 Hi, my exams are over and I have decided to make your request into a stand alone app. I wanted to ask you for any name ideas. Until now I have: BlacklistMonitor, ExcludeMonitor and MonitorLocker. Do you have any additional ideas or which one do you like the most (You can also switch words like MonitorExcluder or ScreenExcluder)? any suggestion is welcome.

**Reply by cevans3505** [reply]
Aghhh, sorry for the late reply. Got your email midday while I was working and meant to circle back to this later and it slipped my mind. If you'd like my input on the naming, I think BlacklistMonitor or ExcludeMonitor give the best idea about the program's intended purpose. MonitorLocker, to me, sounds like it would have more to do with monitor resolution or something.

**Reply by Eliasyoussef47** [reply]
@cevans3505 Just as an update: I haven't forgotten about this, it just turned out more challenging that anticipated. I had made a repo for this project [here](https://github.com/Eliasyoussef47/BlacklistMonitor). I'm also currently busy with a school project so I won't be able to spend a lot of time on this but I'll do my best.

**Reply by cevans3505** [reply]
No worries! I've been getting around the issue by using a spare Raspberry Pi to run that 4th monitor since the information I display on that monitor is static (it's a webpage displaying a phone dashboard). I appreciate you even taking time to work on a solution, but don't feel obligated. If you do end up pursuing it I'll be keeping an eye on the progress. Cheers!
